ramips: add support for Haier HW-L1W

DragonBluep opened the pull request:
Specifications:
* SOC: MT7620A + MT7610E
* ROM: 16 MiB spi flash (Winbond W25Q128)
* RAM: 128MiB DDR2
* WAN: 10/100M *1
* LAN: 10/100M *2
* USB: Type-A USB2.0 *1
* SD: MicroSD *1
* Button: Reset *1
* LEDs: Ethernet *3 + Wlan *1 + USB *1
* Antennas: 2.4 GHz *2 (inside) + 5 GHz *1
* TTL Baudrate: 57600
* Power: DC 5V 2A

Partitions:
mtd0    0x000000000000-0x000001000000 : "ALL"
mtd1    0x000000000000-0x000000030000 : "Bootloader"
mtd2    0x000000030000-0x000000040000 : "Parameter2"
mtd3    0x000000040000-0x000000050000 : "Factory"
mtd4    0x000000050000-0x0000001ed6bb : "Kernel"
mtd5    0x0000001ed6bb-0x000000b50000 : "RootFS"
mtd6    0x000000050000-0x000000b50000 : "Kernel_RootFS"
mtd7    0x000000b50000-0x000000bb0000 : "Kernel_mini"
mtd8    0x000000bb0000-0x000000c00000 : "Parameter1"
mtd9    0x000000c00000-0x000001000000 : "Application"
(Useful data is stored in "Factory" and "Bootloader" so other partitions can be
formated)

TFTP Recovery Info:
* IP: 192.168.1.6, Server: 192.168.1.33, bootfile: ra288.bin
* Push reset button and plug in to upload official firmware ra288.bin.

Installation:
* Reset the router and use putty to open telnet (IP: 192.168.1.1, Port: 20023,
  account: admin, password: admin).
* Backup full flash storage to computer (optional but recommended)
  / $ dd if=/dev/mtd0 of=/tmp/full.bin
  / $ tftp -p -l /tmp/full.bin -r full.bin 192.168.1.x (your IP)
* Use wget command to download openwrt image from local http server to /tmp:
  / $ wget -P /tmp http://192.168.1.x/openwrt-xxxx.bin
* Use mtd_write and dd commands to write it to firmware partitions:
  / $ mtd_write erase mtd6
  / $ dd if=/tmp/openwrt-xxxx.bin of=dev/mtd6

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep commented:
drop useless definitions in "&ethernet", without these definitions ethernet leds can work well as usual.
```
	pinctrl-names = "default";
	pinctrl-0 = <&ephy_pins>;
```

DragonBluep commented:
Re-add pinctrl
```
	pinctrl-names = "default";
	pinctrl-0 = <&ephy_pins>;
```

adschm commented on target/linux/ramips/dts/mt7620a_haier_hw-l1w.dts:
```diff
@@ -0,0 +1,138 @@
+	gpio_uartf {
+		groups = "uartf";
+		function = "gpio uartf";
+	};
```
Why not add this to the gpio node.

DragonBluep replied:
Fixed.

adschm commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -439,6 +439,16 @@ define Device/glinet_gl-mt750
+  DEVICE_VENDOR := Haier
+  DEVICE_MODEL := HW-L1W
+  DEVICE_PACKAGES := kmod-mt76x0e kmod-sdhci-mt7620 \
+  kmod-usb-ohci kmod-usb2 kmod-usb-ledtrig-usbport
```
Hanging indent should be one tab.

DragonBluep replied:
Ok, now fixed.

adschm commented on target/linux/ramips/dts/mt7620a_haier_hw-l1w.dts:
```diff
@@ -0,0 +1,144 @@
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x50000 0xfb0000>;
+			};
```
What's with Parameter1 and Application partitions?

DragonBluep replied:
The  "Application" partition is used to store some plugin just like packages in OpenWrt. And for Parameter1 partition, I read factory image by spi and opened it with a hex editor, nothing important here, just a blank partition filled with “0xff”/ "0x00"(I can't remember now). The one thing that I can sure is when I formatted these two partitions, the official firmware can run normally, so they can merge into the firmware partition.

DragonBluep commented:
Rebase master and rename mac address nvmem-cells to "macaddr_lan".

DragonBluep commented:
Add mac explanation  in commit message and rename "macaddr_lan" back to "macaddr_factory_ffe8".

DragonBluep committed:
ramips: add support for Haier HW-L1W

Specifications:
* SOC: MT7620A + MT7610E
* ROM: 16 MiB spi flash (Winbond W25Q128)
* RAM: 128MiB DDR2
* WAN: 10/100M *1
* LAN: 10/100M *2
* USB: Type-A USB2.0 *1
* SD: MicroSD *1
* Button: Reset *1
* LEDs: Ethernet *3 + Wlan *1 + USB *1
* Antennas: 2.4 GHz *2 (inside) + 5 GHz *1
* TTL Baudrate: 57600
* Power: DC 5V 2A

MAC addresses as verified by OEM firmware:
use   address   source
2g    *:c9      factory 0x0004
5g    *:ca      factory 0x8004
LAN   *:c7      factory 0xffee
WAN   *:c8      factory 0xffe8 (label)

Partitions:
mtd0    0x000000000000-0x000001000000 : "ALL"
mtd1    0x000000000000-0x000000030000 : "Bootloader"
mtd2    0x000000030000-0x000000040000 : "Parameter2"
mtd3    0x000000040000-0x000000050000 : "Factory"
mtd4    0x000000050000-0x0000001ed6bb : "Kernel"
mtd5    0x0000001ed6bb-0x000000b50000 : "RootFS"
mtd6    0x000000050000-0x000000b50000 : "Kernel_RootFS"
mtd7    0x000000b50000-0x000000bb0000 : "Kernel_mini"
mtd8    0x000000bb0000-0x000000c00000 : "Parameter1"
mtd9    0x000000c00000-0x000001000000 : "Application"
Useful data is stored in "Factory" and "Bootloader" so other partitions can be
formated.

TFTP Recovery Info:
* IP: 192.168.1.6, Server: 192.168.1.33, bootfile: ra288.bin
* Push reset button and plug in to upload official firmware ra288.bin.

Installation:
* Reset the router and use putty to open telnet (IP: 192.168.1.1, Port: 20023,
  account: admin, password: admin).
* Backup full flash storage to computer (optional but recommended)
  / $ dd if=/dev/mtd0 of=/tmp/full.bin
  / $ tftp -p -l /tmp/full.bin -r full.bin 192.168.1.x (your IP)
* Use wget command to download openwrt image from local http server to /tmp:
  / $ wget -P /tmp http://192.168.1.x/openwrt-xxxx.bin
* Use mtd_write and dd commands to write it to firmware partitions:
  / $ mtd_write erase mtd6
  / $ dd if=/tmp/openwrt-xxxx.bin of=dev/mtd6

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

adschm commented on target/linux/ramips/dts/mt7620a_haier_hw-l1w.dts:
```diff
@@ -0,0 +1,144 @@
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
```
Is Parameter2 really u-boot-env?

adschm commented on target/linux/ramips/dts/mt7620a_haier_hw-l1w.dts:
```diff
@@ -0,0 +1,144 @@
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
```
Is Bootloader really u-boot?

adschm commented:
LAN/WAN MAC addresses are swapped when comparing commit message and actual setup.

DragonBluep commented:
Finally, I found that the original uboot needs a suitable firmware header. Thanks for reviewing, maybe it needs to be temporarily closed.
